src/Attributable.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace SilverStripe\Assets;
+
+use SilverStripe\Core\Convert;
+
+trait Attributable
+{
+    protected $attributes = [];
+
+//    private static $casting = [
+//        'AttributesHTML' => 'HTMLFragment',
+//        'getAttributesHTML' => 'HTMLFragment',
+//    ];
+
+
+
+    public function setAttribute($name, $value)
+    {
+        $this->attributes[$name] = $value;
+
+        return $this;
+    }
+
+    public function getAttribute($name)
+    {
+        $attributes = $this->getAttributes();
+
+        if (isset($attributes[$name])) {
+            return $attributes[$name];
+        }
+
+        return null;
+    }
+
+    protected abstract function getDefaultAttributes(): array;
+
+    /**
+     * Allows customization through an 'updateAttributes' hook on the base class.
+     * Existing attributes are passed in as the first argument and can be manipulated,
+     * but any attributes added through a subclass implementation won't be included.
+     *
+     * @return array
+     */
+    public function getAttributes()
+    {
+        $defaultAttributes = $this->getDefaultAttributes();
+
+        $attributes = array_merge($defaultAttributes, $this->attributes);
+
+        if (method_exists($this, 'extend')) {
```

Comment:
It would be fair to assume that which ever class implements this must also implements Extensible, but by have a function check here, we don't force it.

Reply:
When we go with option 1 there's probably extra code like this that's currently in the trait that can be removed